File: Python/Fundamentals/Data_Structures/Singly_Linked_List/singly_linked_list.py
class Node():
    def __init__(self, value):
        self.value = value
        self.next = None


class Singly_Linked_List:
    def __init__(self):
        self.head = None

    def add_first(self, value):
        new_node = Node(value)
        current_head = self.head
        new_node.next = current_head
        self.head = new_node
        return self

    def push(self, value):
        new_node = Node(value)
        iterator = self.head
        while iterator.next != None:
            iterator = iterator.next
        iterator.next = new_node
        return self

    def remove_front(self):
        self.head = self.head.next
        return self

    def remove_back(self):
        pass

    def print_values(self):
        iterator = self.head
        while iterator != None:
            print(iterator.value)
            iterator = iterator.next
        return self


sl_list = Singly_Linked_List()
sl_list.add_first(1).push(2).push(
    3).print_values().remove_front().print_values()

print('')
print('Head:')
print(sl_list.head.value)

File: Java/Fundamentals/OOP/sll/SinglyLinkedList.java
public class SinglyLinkedList {
    public Node head;
    
    public SinglyLinkedList() {
        this.head = null;
    }

    public SinglyLinkedList(int headValue) {
        Node headNode = new Node(headValue);
        this.head = headNode;
    }

    public void add(int value) {
        Node newNode = new Node(value); 
        if (this.head == null) {
            this.head = newNode;
            return;
        }
        Node lastNode = this.head;
        while (lastNode.next != null) {
            lastNode = lastNode.next;
        }
        lastNode.next = newNode;
    }

    public void removeLast() {
        Node runner = this.head;
        while (runner.next.next != null) {
            runner = runner.next;
        }
        runner.next = null;
    }

    public void printValues() {
        Node runner = this.head;
        while (true) {
            System.out.println(runner);
            if (runner.next == null) {
                break;
            }
            runner = runner.next;
        }
    }

    public Node find(int value) {
        Node runner = this.head;
        while (runner.next != null) {
            if (runner.value == value) {
                return runner;
            }
            runner = runner.next;
        }
        return null;
    }

    public void removeAt(int position) {
        if (position == 0) {
            this.head = this.head.next;
            return;
        }
        Node previousNode = this.head;
        Node currentNode = this.head.next;
        for (int i=1; i<=position; i++) {
            if (i == position) {
                previousNode.next = currentNode.next;
                break;
            }
            previousNode = currentNode;
            currentNode = currentNode.next;
        }
    }
}